### DISCUSSION TITLE
Tailwind ignore content config, why though?!

### DISCUSSION THREAD

**Discussion by J4v4Scr1pt** [tailwindlabs/tailwindcss#12366] [Help] [answered]
I have created my own component lib in storybook using Tailwind. When I use this inside my application I don't want to include some generated css file from the lib. I just want tailwind to scan my components in the node_modules, like NextUI does it and add used classes. NextUI works like a charm and if I remove it from content config styles get removed from the UI, Beautiful!

```
content: [
    './components/**/*.{js,ts,jsx,tsx,mdx}',
    './app/**/*.{js,ts,jsx,tsx,mdx}',
    './node_modules/@my/components/dist/components/**/*.{js,ts,jsx,tsx}',
    './node_modules/@nextui-org/theme/dist/**/*.{js,ts,jsx,tsx}',
],
```

I have tried changing the path to:
`'./node_modules/@my/components/dist/components/Button/Button.js',`
`'./node_modules/@my/**/*.{js,ts,jsx,tsx}',`

But no styles applied...

This is the structure of my lib:
![image](https://github.com/tailwindlabs/tailwindcss/assets/5929055/51a2aed4-91f4-45b3-8d79-b46c4adf7861)

This is the file in the package:
```
import { jsxs as i } from "react/jsx-runtime";
import { tv as m } from "tailwind-variants";
const n = m({
  base: "font-medium bg-blue-500 text-white rounded-full active:opacity-80",
  variants: {
    color: {
      primary: "bg-blue-500 text-white",
      secondary: "bg-purple-500 text-white",
      secondaryTwo: "bg-red-500 text-black"
    },
    size: {
      sm: "text-sm",
      md: "text-base",
      lg: "px-4 py-3 text-lg"
    }
  },
  compoundVariants: [
    {
      size: ["sm", "md"],
      class: "px-3 py-1"
    }
  ],
  defaultVariants: {
    size: "md",
    color: "primary"
  }
}), l = ({ size: e, className: o, color: r, label: t, children: s, ref: a }) => /* @__PURE__ */ i(
  "button",
  {
    ref: a,
    type: "button",
    title: t,
    className: n({ className: o, color: r, size: e }),
    children: [
      s,
      t
    ]
  }
);
export {
  l as Button
};
```

I would love some help to understand why this is ignored by Tailwind. Probably I do something wrong here.. 😅

**Comment by alex-krasikau** [comment]
Hey @J4v4Scr1pt!

The setup seems right. Let's try a couple of steps to identify the cause of this issue.


1. First, we can try the broadest content config to make the library styles work in your project:
```js
content: [
    './components/**/*.{js,ts,jsx,tsx,mdx}',
    './app/**/*.{js,ts,jsx,tsx,mdx}',
    './node_modules/**/*.{js,ts,jsx,tsx}',
],
```

With this configuration, we anticipate one of these four outcomes:
- Library and NextUI applying styles
- Only NextUI applying styles, but the Library not functioning
- Library applying styles, but NextUI not
- Neither Library nor NextUI applying styles

Please inform me of the results.


2. Although the configuration appears correct, let's verify if you're correctly using the component in your code.
- It would be excellent if you could share the resulting Tailwind CSS output.css file.
- Could you please provide a screenshot of the Library button in the Browser inspector?

**Reply by J4v4Scr1pt** [reply]
Yes will get back to you asap!

**Reply by J4v4Scr1pt** [reply]
After I provide the answer I noticed that the thing that is missed is the bg-purple-500, if I changed to another it do not work either..

I then removed our theme colors and it now works, guess it's how i declare the colors and they dont have the -500 tone.

```javascript
theme: {
		colors: {
			purple: '#7f2696',
			darkPurple: '#4e0854',
			lightPurple: '#f4efff',
			whitePurple: '#f9f7ff',
			alternativePurple: '#e2d6ff',
			alternativeLightPurple: '#EEE6FF',
			pink: '#c86ee5',
			turquoise: '#00d2bf',
			jungle: '#00bdab',
			skobeloff: '#007374',
			lightTurquoise: '#c6f4ec',
			whiteTurquoise: '#f1fcfa',
			red: '#ff444c',
			coral: '#ff6f79',
			orange: '#ffa000',
			honey: '#906C00',
			yellow: '#ffc107',
			lightYellow: '#feeec2',
			whiteYellow: '#fff8e6',
			black: '#000000',
			white: '#ffffff',
			redOchre: '#ee6c4d',
			brick: '#b63c23',
			lightRed: '#ffe8ea',
			whiteRed: '#fff9f7',
			alternativeRed: '#fff5f6',
			palmLeaf: '#6da34d',
			green: '#71bc49',
			sapGreen: '#3e7421',
			pistachio: '#9ed57b',
			royalBlue: '#4764c5',
			cobalt: '#213d98',
			oceanBlue: '#3e7b8b',
			lightBlue: '#CBF4FF',
			whiteBlue: '#ECFBFF',
			skyBlue: '#59b0c7',
			lightCornflower: '#f6f7fc',
			bronze: '#cf8426',
			gambodge: '#995700',
			pastelOrange: '#ffb456',
			onyx: '#353a40',
			nickel: '#6b6f75',
			metal: '#cccccc',
			fog: '#eeeeee',
			smoke: '#f5f5f5',
			whiteSmoke: '#fafafa',
			vibrantRed: '#ff5c61',
			vibrantOrange: '#ffa000',
			vibrantYellow: '#ffd900',
			vibrantLime: '#B3E818',
			vibrantGreen: '#45CB1C',
			lightOrange: '#ffe2c1',
			darkOrange: '#8D4D1A',
			whiteOrange: '#FFF5E5',
			blue: '#05BDEE',
			darkBlue: '#03718E',
			lightVibrantRed: '#FFEFEF',
			lightVibrantYellow: '#FFFADC',
			lightVibrantOrange: '#FFF5E5',
			lightVibrantLime: '#F7FFE0',
			lightVibrantGreen: '#EBFFE4',
			misteryBlack: '#181A1E',
			sand: '#F4F2F0',
		},
		fontFamily: {
			ttCommonsRegular: ['var(--font-TT-commons-regular)'],
			ttCommonsDemibold: ['var(--font-TT-commons-demibold)'],
			ttCommonsItalic: ['var(--font-TT-commons-italic)'],
			ttCommonsMedium: ['var(--font-TT-commons-medium)'],
		},
		extend: {
			gridTemplateColumns: {
				// Complex site-specific column configuration
				header: '72px calc(100vh - 72px) / max-content',
			},
		},
	},
```

**Reply by J4v4Scr1pt** [reply]
I tested to prefix the colors with `my_purple` etc. But it did not work. I need to remove the color setting completely for the colors to apply correctly. Is this correct behavior when using the color setting?

**Reply by J4v4Scr1pt** [reply, marked as answer]
You can add me to the stupid list.. 
I forgot to add the colors in the extend.... 😭.

Thanks for the quick awnser!

Take care!

**Reply by alex-krasikau** [reply]
I'm glad you found the solution. I typically don't regret these situations because today, you've become more experienced than you were three hours ago. 😉

**Reply by J4v4Scr1pt** [reply]
Holly jesus @alex-krasikau I wish all people had your mindset! 
All best to you and keep up the awesome work!

**Comment by J4v4Scr1pt** [comment]
Thanks for getting back to me this fast! After changing to your suggested content I restared NextJs and did a hard refresh.
Then the intial load took 3.5min to load, guess that is because of the scan throug all node_modules.

- NextUI still worked but my component still did not. 

This is the html from the button in the inspector:
```html
<button type="button" title="TEST" class="font-medium rounded-full active:opacity-80 bg-purple-500 text-white text-base px-3 py-1">TEST</button>
``` 
Underneth my TEST button are the NextUI components I just threw in to test.
![image](https://github.com/tailwindlabs/tailwindcss/assets/5929055/5c9276fc-23df-4363-90ce-d7e01a800e43)


Tailwind css output (It became quite loong.. 😅):

```css
REMOVED
``` 


Edit: since the issue was not because of tailwind I removed the generated CSS because it was slowing down the page and was to large
